servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpMessageDiscardWatchdogClientFilter.java

```diff
@@ -107,7 +107,7 @@ protected Single<StreamingHttpResponse> request(final StreamingHttpRequester del
                                                                 final StreamingHttpRequest request) {
                     return delegate
                             .request(request)
-                            .onErrorResume(cause -> {
+                            .whenOnError(cause -> {
```

Comment:
ffti: if we want to preserve the 'sequencing' of events, maybe we should be explicit and use `.beforeOnError`. I recognize that is what `.whenOnError` forwards too, but technically the `whenOnError` behavior is undefined and supposedly could change.